include/ac_npc_shop_master.h

```diff
@@ -3,16 +3,265 @@
+    aNSC_MSG_SHOW_CLOTH = 0x12,
+    aNSC_MSG_CONFIRM_CLOTH = 0x13,
+    aNSC_MSG_CANCEL_CLOTH = 0x14,
+    aNSC_MSG_CANCEL = 0x15,
+    aNSC_MSG_START_CALL_NORMAL = 0x16,
+    aNSC_MSG_INTERACT_START = 0x17,
+    aNSC_MSG_INTERACT_CANCEl = 0x18,
+    aNSC_MSG_KABU_INFO = 0x19,
+    aNSC_MSG_SELL_START = 0x1a,
+    aNSC_MSG_SELL_OFFER = 0x1b,
+    aNSC_MSG_SELL_NORMAL = 0x1c,
+    aNSC_MSG_SELL_CANCEL = 0x1d,
+    aNSC_MSG_BUY_ONE_OFFER = 0x1e,
+    aNSC_MSG_BUY_CANCEL = 0x1f,
+    aNSC_MSG_THANKS = 0x20,
+    aNSC_MSG_INSUFFICIENT_FUNDS = 0x21,
+    aNSC_MSG_BUY_NORMAL = 0x22,
+    aNSC_MSG_SAY_GOODBYE = 0x23,
+    aNSC_MSG_KABU_ON_SUNDAY = 0x24,
+    aNSC_MSG_EVENT = 0x25,
+    aNSC_MSG_BREAK_BAG = 0x26,
+    aNSC_MSG_POCKETS_FULL = 0x27,
+    aNSC_MSG_SHOW_ITEM = 0x28,
+    aNSC_MSG_SHOW_CARPET_OFFER = 0x29,
+    aNSC_MSG_SHOW_CARPET_DECLINE = 0x2a,
+    aNSC_MSG_SHOW_CATALOGUE = 0x2b,
+    aNSC_MSG_ORDER_OFFER = 0x2c,
+    aNSC_MSG_ORDER_UNAVAILABLE = 0x2d,
+    aNSC_MSG_ORDER_CANCEL = 0x2e,
+    aNSC_MSG_ORDER_CONFIRM = 0x2f,
+    aNSC_MSG_ORDER_INSUFFICIENT_FUNDS = 0x30,
+    aNSC_MSG_ORDER_FULL = 0x31,
+    aNSC_MSG_ORDER_FURNITURE_ONLY = 0x32,
+    aNSC_MSG_GIVE_TICKET = 0x33,
+    aNSC_MSG_MAIL_TICKET = 0x34,
+    aNSC_MSG_TICKET_1 = 0x35,
+    aNSC_MSG_TICKET_2 = 0x36,
+    aNSC_MSG_TICKET_SELL_REFUSE = 0x37,
+    aNSC_MSG_SELL_QUEST = 0x38,
+    aNSC_MSG_JUNK_ACCEPT = 0x39,
+    aNSC_MSG_JUNK_CANCEL = 0x3a,
+    aNSC_MSG_SELL_SHOVEL = 0x64,
+    aNSC_MSG_SELL_NET = 0x65,
+    aNSC_MSG_SELL_ROD = 0x66,
+    aNSC_MSG_SELL_AXE = 0x67,
+    aNSC_MSG_SELL_SIGN = 0x68,
+    aNSC_MSG_MONEY_OVERFLOW = 0xc8,
+    aNSC_MSG_SELL_PAINT = 0xca,
+    ansc_msg_sell_paint_confirm = 0xcb,
+    aNSC_MSG_BASEMENT_BUILT = 0xcc,
+    aNSC_MSG_BASEMENT_PAID = 0xcd,
+    aNSC_MSG_REHOUSE_BASEMENT = 0xd0,
+    aNSC_MSG_REHOUSE_LARGE = 0xd1,
+    aNSC_MSG_REHOUSE_UPPER = 0xd4,
+    aNSC_MSG_UPPER_BUILT = 0xd5,
+    aNSC_MSG_BUY_MANY_OFFER = 0xd6,
+    aNSC_MSG_BUY_REFUSE = 0xd8,
+    aNSC_MSG_BALLOON = 0xda,
+    aNSC_MSG_BALLOON_DEPART = 0xdc,
+    aNSC_MSG_SELL_PAPER = 0xde,
+    aNSC_MSG_SELL_PAINT_FOREIGN = 0xdf,
+    aNSC_MSG_PSWD_NO_ITEM = 0x12c,
+    aNSC_MSG_PSWD_INFO = 0x12d,
+    aNSC_MSG_PSWD_SAME_NAME = 0x12e,
+    aNSC_MSG_PSWD_CONFIRM = 0x12f,
+    aNSC_MSG_PSWD_CANCEL = 0x131,
+    aNSC_MSG_PSWD_CHECK = 0x132,
+    aNSC_MSG_OTHER = 0x135,
+    aNSC_MSG_RETURN = 0x136,
+    aNSC_MSG_PSWD_INVENTORY_FULL = 0x137,
+    aNSC_MSG_PSWD_SAY = 0x138,
+    aNSC_MSG_PSWD_WRONG = 0x139,
+    aNSC_MSG_PSWD_DECLINE = 0x13a,
+    aNSC_MSG_PSWD_GOOD_FAMICON = 0x13b,
+    aNSC_MSG_PSWD_GOOD_NPC = 0x13c,
+    aNSC_MSG_PSWD_GOOD_MAGAZINE_1 = 0x13d,
+    aNSC_MSG_PSWD_CARDE = 0x13e,
+    aNSC_MSG_PSWD_GOOD_USER = 0x13f,
+    aNSC_MSG_PSWD_GOOD_CARDEM = 0x140,
+    aNSC_MSG_PSWD_GOOD_MAGAZINE_2 = 0x142,
+    aNSC_MSG_PSWD_BAD = 0x144,
+    aNSC_MSG_PSWD_FOREIGN = 0x141,
+    aNSC_MSG_GIFT_OUT = 0x143,
+    aNSC_MSG_TAKE_OFF_HANDS= 0x108a,
```

Comment:
This one appears to be the literal message ID instead of a relative value. It shouldn't be in this enum. You can add a definition for it in `m_string_data.h` if you'd like.

Reply:
They sort of are relative values; IDs 0-99 have one offset, IDs 100-199 are another, etc. (See https://github.com/colbyaustinbrown/ac-decomp/blob/1b4d4894b325ff1193276b6896d2701b4437a3bf/src/actor/npc/ac_npc_shop_master.c#L895). I can move them if you like, but thought I would give the extra context before I do.